libEDSsharp/eds.cs

```diff
@@ -679,7 +679,7 @@ public partial class DeviceInfo : InfoSection
         /// product revision number according to identity object sub-index 03h (Unsigned32) 
         /// </summary>
         [EdsExport]
-        public UInt32 RevisionNumber;
+        public string RevisionNumber="";
```

Comment:
RevisionNumber is like ProductNumber, I used string because it can be represented as both decimal and hexadecimal

Reply:
The tool always uses the string type for this data, the RevisionNumber (must have the major equal to OD 1018 subindex 3) was always set by default and could not be changed from the graphical interface

Reply:
> RevisionNumber is like ProductNumber, I used string because it can be represented as both decimal and hexadecimal

I see, but how can we make sure it only contains a valid number and not a string?

Reply:
The tool does not check this in any object, you can try for example with the ProductID, DeviceType, etc...
I changed the RevisionNumber to a string to make it conform to the rest of the existing code.

To solve your question, you should create a universal function that does this check and attach it to all objects, but it requires a major change (but this was not my goal)

Reply:
I prefer to stay uint32 in database. GUI should be able to represent it as both decimal and hexadecimal.
However, `ProductNumber` should also be `uint32`...

See also our new ideas for those issues in #115